core/src/main/java/org/springframework/security/access/expression/method/DefaultMethodSecurityExpressionHandler.java

```diff
@@ -172,8 +173,27 @@ public Object filter(Object filterTarget, Expression filterExpression,
 			return filtered;
 		}
 
+		if (filterTarget instanceof Stream) {
+			final Stream<?> original = (Stream<?>) filterTarget;
+			if (debug) {
+				logger.debug("Filtering stream with " + original.count() + " elements");
```

Comment:
While I appreciate the consistency that this affords to the other two data types this class supports, this won't work since you cannot operate on a stream after it's been terminated.

e.g. after running line 179, then line 182 would fail with `stream has already been operated on or closed`.

This would be the same with line 189 and whatever the user wanted to do downstream.

That being the case, let's remove these debug if blocks.